jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-backend/src/main/java/org/jbpm/workbench/cm/backend/server/RemoteCaseManagementServiceImpl.java

```diff
@@ -133,9 +134,51 @@ public void assignGroupToRole(final String serverTemplateId, final String contai
     }
 
     @Override
-    public void assignGroupAndUserToRole(final String serverTemplateId, final String containerId, final String caseId, final String roleName, final String user, final String group) {
-        client.assignGroupToRole(containerId, caseId, roleName, group);
-        client.assignUserToRole(containerId, caseId, roleName, user);
+    public void assignToRole(final String serverTemplateId, final String containerId, final String caseId,
+                             final String roleName, final List<String> usersList, final List<String> groupsList) {
+        CaseRoleAssignmentSummary ras = getCaseInstance(serverTemplateId, containerId, caseId).getRoleAssignments()
+                .stream()
+                .filter(ra -> roleName.equals(ra.getName()))
+                .findFirst().orElse(null);
+
+        if (ras != null) {
+            if (ras.getUsers() != null) {
+                List<String> usersToremove = new ArrayList();
+                ras.getUsers().forEach(
+                        user -> {
+                            if (!usersList.stream().filter(newU -> user.equals(newU)).findFirst().isPresent()){
+                                usersToremove.add(user);
+                            }
+                        });
+                usersToremove.forEach(user->removeUserFromRole(serverTemplateId, containerId, caseId, roleName, user));
+                usersList.forEach(
+                        newUser -> {
+                            if (!ras.getUsers().stream().filter(user -> newUser.equals(user)).findFirst().isPresent()){
+                                assignUserToRole(serverTemplateId, containerId, caseId, roleName, newUser);
+                            }
+                        });
+            }
+            if (ras.getGroups() != null) {
+                List<String> groupsToremove = new ArrayList();
+                ras.getGroups().forEach(
+                        group -> {
+                            if (!groupsList.stream().filter(newGroup -> group.equals(newGroup)).findFirst().isPresent()){
+                                groupsToremove.add(group);
+                            }
+                        });
+                groupsToremove.forEach(group ->removeGroupFromRole(serverTemplateId, containerId, caseId, roleName, group));
+                groupsList.forEach(
+                        newGroup -> {
+                            if (!ras.getGroups().stream().filter(group -> newGroup.equals(group)).findFirst().isPresent()){
+                                assignGroupToRole(serverTemplateId, containerId, caseId, roleName, newGroup);
+                            }
+                        });
+            }
+        } else {
+           usersList.forEach(u -> assignUserToRole(serverTemplateId, containerId, caseId, roleName, u));
+            groupsList.forEach(g -> assignGroupToRole(serverTemplateId, containerId, caseId, roleName, g));
```

Comment:
This approach allows entering duplicate users/groups, as _CommaListValuesConverter_ doesn't perform any filtering if duplicates are entered.